packages/server/graphql/mutations/helpers/notifySlack.ts

```diff
@@ -97,11 +104,54 @@ export const startSlackMeeting = async (
     utm_campaign: 'invitations'
   }
   const options = {searchParams}
-  const team = await dataLoader.get('teams').load(teamId)
-
+  const [team, meeting] = await Promise.all([
+    dataLoader.get('teams').load(teamId),
+    dataLoader.get('newMeetings').load(meetingId)
+  ])
   const meetingUrl = makeAppURL(appOrigin, `meet/${meetingId}`, options)
-  const slackText = `${team.name} has started a meeting!\n To join, click here: ${meetingUrl}`
-  notifySlack('meetingStart', dataLoader, teamId, slackText).catch(console.log)
+  const button = {text: 'Join meeting', url: meetingUrl, type: 'primary'} as const
+  const blocks = [
+    makeSection('Meeting started :wave: '),
+    makeSections([`*Team:*\n${team.name}`, `*Meeting:*\n${meeting.name}`]),
+    makeSection(`*Link:*\n<${meetingUrl}|https:/prbl.in/${meetingId}>`),
+    makeButtons([button])
+  ]
+  notifySlack('meetingStart', dataLoader, teamId, blocks).catch(console.log)
+}
+
+const getSummaryText = (meeting: MeetingRetrospective | MeetingAction | MeetingPoker) => {
+  if ('reflectionCount' in meeting) {
```

Comment:
for discriminated unions like this it's a little cleaner to switch on the type
you can use a switch or a condition, both give full type safety as long as you use object access & that way if reflectionCount goes away, we don't have to change the logic:

```ts
  switch (meeting.meetingType) {
    case 'retrospective':
      
  }
  if (meeting.meetingType === 'retrospective') {

  }

Reply:
I tried to use the meeting type initially but it means we have to cast the type (which we do in [joinMeeting](https://github.com/ParabolInc/parabol/blob/4ce15d9ba712b4004d6361be3c5daee5714e0ad9/packages/server/graphql/mutations/joinMeeting.ts#L29)). We don't need to cast the type when using  `reflectionCount` but I definitely agree that using a field that could be removed at some point isn't ideal

Reply:
well then i made those classes poorly.
For each meeting class (e.g. MeetingRetrospective) could you add the meeting type as a string const?
for example

```ts
export default class MeetingRetrospective extends Meeting {
  meetingType!: 'retrospective'
}
```

after you've done that for all 3 you'll be able to discriminate on the value of `meetingType`

Reply:
Cool thanks - I've pushed that update and it's working nicely